### PR TITLE
Kills deprecated code and fixes sonarlint inspections.

### PR DESCRIPTION
**Pull request by andyHa**
Should probably be merged together with:
* https://github.com/scireum/sirius-kernel/pull/404
* https://github.com/scireum/sirius-web/pull/1101
* https://github.com/scireum/sirius-db/pull/522

### CODE REVIEW

File: src/main/java/sirius/biz/tycho/kb/KBHelper.java [hunk 1]
```diff
@@ -31,4 +31,8 @@ public KnowledgeBaseArticle currentArticle() {
     protected void installCurrentArticle(KnowledgeBaseArticle article) {
         currentArticle.set(article);
     }
+
+    protected void clearCurrentArticle() {
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/tycho/kb/KBHelper.java:35]: Javadoc is missing.

**Review by mkeckmkeck — APPROVED** [review]

**Review by mko-sci — APPROVED** [review]

File: src/main/java/sirius/biz/model/ContactData.java [hunk 2]
```diff
@@ -108,7 +119,8 @@ public class ContactData extends Composite {
      * @throws HandledException in case of an invalid phone number
      */
     public void verifyPhoneNumber() {
-        if (Strings.isFilled(phone) && !VALID_PHONE_NUMBER.matcher(phone).matches()) {
+        if (Strings.isFilled(phone) && phone.length() < MAX_PHONE_NUMBER_LENGTH && !VALID_PHONE_NUMBER.matcher(phone)
```
Comment (sabieber) [inline src/main/java/sirius/biz/model/ContactData.java:122]: So an entered phone number that is 256 or longer is automatically valid?

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/tycho/kb/KBHelper.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/biz/tycho/kb/KBHelper.java:45]: Technically protected ;-) - but still documented now^^

File: src/main/java/sirius/biz/model/ContactData.java (hunk 2 above)
Comment (andyHa) [reply src/main/java/sirius/biz/model/ContactData.java]: The check has ben overhauled and now uses a normal regex with possesive quantifiers.

**Review by qw3ry — APPROVED** [review]

**Pull request merged**